### ISSUE TITLE
Windurst Mission 4-1 (Magicite) Mission Turn in not working

### ISSUE THREAD

**Issue by aprilgarrity** [EdenServer/community#2318]
### Checklist

- [x] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
- [x] I have provided reproducable steps. (No "as the title says" posts please. Provide reproducable steps even if it seems like a waste of time.)
- [x] I have provided my client version in the details. (type /ver into your game window)

### Details
VERSION: 30181205_0

I obtained the items needed for the Magicite mission (Orastone, Aurastone, and Opistone) via steps outlined on the wiki, and successfully returned to the Audience Chamber in Ru'Lude Gardens, where I obtained an airship pass. 

The final step of the mission involves returning to the receptionist (Pakh Jatalfih at I-9 in Ru'Lude Gardens); however, when I do so, she gives me the line of dialogue: "As long as the Star Sibyl is safe, then all is well. We don't have any tasks for you here, so continue your service to your country." She is supposed to give me the key item "Message to Jeuno," 10,000 gil, and promote me to rank 5, and immediately start the 5-1 mission. However, none of these things happen and she gives me the dialogue line stated above.

I have been repeatedly talking to her for days after relaunching the game, restarting my computer, zoning, doing entire quests and then returning, etc. My Rank Points bar is full and I have double-checked with the conquest guard; he tells me "You do not need to donate any more crystals at your current rank. Speak to the guard at the gatehouse for further instructions." 

I am not sure what else it could be at this point, but no one else has had an issue with this, so I have been hesitant to submit a ticket; however, since it has been several days, I figured I would do so anyway.

Another odd note: When I speak to the secretaries of the other two nations in Ru'Lude Gardens, I get the magicite cutscenes for those nations. When the cutscene ends, I'm in a black room with only the NPC and the door behind me. When I run to the door and open it, the game returns to normal. The cutscene claims I obtain a new Audience Permit, but when I check my key items, it is not there. This occurred both before and after I obtained the airship pass.

**Comment by aprilgarrity** [comment]
My IGN is PanickingPig, if that is needed.

**Comment by Althaia** [comment]
That is definitely what is supposed to happen when you return to the secretary... It's been so long since I obtained Rank 5 that I don't know if I'll be much help here...

But in the interest of hopefully providing a solution, have you tried talking to the Ambassador in the back/side room? I seem to recall I was following my friend at the time and we both sort of assumed we had to talk to the Ambassador, not the secretary.

So we talked to the ambassador, then the secretary, and then got Rank 5. I think. Maybe try that?

Were you doing Magicite with other people who were also doing it for the first time? If so, did they get Rank 5 successfully? Also if so, from which nations were they?

Would be weird if something got screwed up for Windurstians in an update at some point, though I do seem to recall people (non-Windurstians?) asking for 5-1 / 5-2 help recently... And Mercury was doing 5-1/5-2 runs specifically for Windurstians, which I think was either in October or November... I can ask Mercury for more info if you want.

**Comment by aprilgarrity** [comment]
I have tried going in to the backroom, but I am not granted access. Instead, I overhear a conversation that goes: "I have a letter for your here from none other than Atori-Tutori. It reads, ahem..." And the door does not open, it does not state an NPC that is talking, nothing.

I was doing the missions with two other people, but I do not know what nations they were from; they did not report any issues to me.

**Comment by aprilgarrity** [comment]
I am thinking that is might be bugged for me because of the additional note I added in my original post; the fact that I get the same cut-scene over and over again in both the Bastok and Sandy embassies is really strange, and then I'm left in a black room with just the receptionist and the outside door; I'll attach a screenshot of this.
![cutscene](https://user-images.githubusercontent.com/28421929/71600805-9ceccf80-2b1e-11ea-9ad3-a3ef438936a4.png)
![black room](https://user-images.githubusercontent.com/28421929/71600814-a37b4700-2b1e-11ea-9efc-9430e081c171.png)

**Comment by MogwaiFFXI** [comment]
I have this very same issue and I am blocked from obtaining Rank 5.

I received the airship pass and was told to return to my embassy secretary Pakh Jatalfih.

If it helps I did accidentally talk to the Bastok receptionist Goggehn instead. I received the same repeated cutscene and effects described by the other posters.

Now I am soft locked at rank 4 and cannot salk to Pakh Jatalfih to finish 4-1.

**Comment by Althaia** [comment]
@MogwaiFFXI I know Lorelai and I'm pretty sure she's past Rank 5 now, so you may want to try to talk to her in-game to see what she did to fix it. I think she had to talk to a GM and get her Rank 5 manually? Maybe try that, too.

**Comment by MogwaiFFXI** [comment]
@Althaia Thanks I've reached out to Lorelai and they confirmed they were manually progressed through the mission. The underlying bug could be that you can interact with an embassy other than your own and trigger the opening cutscene again. Only those that make the mistake of talking to the wrong embassy would get the bug.

**Comment by MogwaiFFXI** [comment]
For the records a GM reset my mission back to 4-1 Magicite. As I already had all key items for accessing the Beastman strongholds I just travelled around and got all the Magicite cutscenes again.

This time I returned to the palace and to the correct Windurst secretary and the mission completed successfully. Now rank 5, mildly annoying having to travel all over again but happy I can progress again.

**Comment by EdenPenance** [comment]
Ran into this again today. 

Rolling back DSP code to before our last pull it looks like if you accidentally talk to the wrong Embassy NPCs in Ru'Lude Gardens while on Rank 4, it can reset your Rank 4 mission progress. If you talk to the Windy Embassy NPC, it'll give you the wrong messages but make no variable changes. If you weren't from Sandy/Bastok and talked to those NPCs, it'll reset your Rank 4 mission progress.

**tl;dr:**
The player:getCurrentMission(SANDORIA) == 255 and player:getCurrentMission(Bastok) == 255 checks in Goggehn/Nelcabrit's .lua should probably be changed to a general "what nation is the player" check, and moved to the bottom of the magicite checks, but above the MESSAGE_TO_JEUNO related checks. Alternatively, this is the commit that fixed the issue on DSP: https://github.com/DarkstarProject/darkstar/commit/202c612f35752439cf3c0d173335f9e2aac0ee9c#diff-2310b9f2cf2d1e30a3063b9a2df025eda5ee9d162958908e2345071fc19939b6

Link to backdated code: https://github.com/DarkstarProject/darkstar/tree/14223ce7a31515db7494aec7beb5f8b25e8b04fd/scripts/zones/RuLude_Gardens/npcs

Goggehn's .lua:  
![image](https://user-images.githubusercontent.com/76273942/110260040-3a882e00-7f70-11eb-80e3-2044085184a8.png)

Nelcabrit's .lua:
![image](https://user-images.githubusercontent.com/76273942/110260064-4ecc2b00-7f70-11eb-96dc-411fd7ce8080.png)

The red-boxed sections are what's causing the problem. It does a check to see if you have Bastok/Sandy mission 255 which is "NONE". As a non-Bastok/Sandy person, that's what your mission ID is for those nations so it passes the check there. That makes it go to CS129 (Bastok) or CS130 (Sandy) which both do player:setVar("MissionStatus",1); and player:addKeyItem(dsp.ki.ARCHDUCAL_AUDIENCE_PERMIT); (if you don't already have the KI).

**Comment by 59blargedy** [comment]
resolved with April 26, 2021 update
